src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -152,7 +159,21 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
             {
                 var config = _analyzerConfigs[analyzerConfigIndex];
 
-                if (normalizedPath.StartsWith(config.NormalizedDirectory, StringComparison.Ordinal))
+                if (config.IsGlobal)
+                {
+                    // We treat global configs differently:
+                    // - global sections are included
+                    // - section names must be full path exact matches
+                    sectionKey.Add(config.GlobalSection);
+                    foreach (var section in config.NamedSections)
+                    {
+                        if (normalizedPath.Equals(section.Name, StringComparison.Ordinal))
```

Comment:
We should re-use the `Section.NameComparer` field here vs. `StringComparison.Ordinal`. That is the central field that controls how these names are compared.